Classes/Common/AbstractPlugin.php

```diff
@@ -198,40 +199,68 @@ protected function loadDocument()
     abstract public function main($content, $conf);
 
     /**
-     * Wraps the input string in a tag with the class attribute set to the class name
+     * Parses a string into a Typoscript array
+     *
+     * @access protected
+     *
+     * @param string $string: The string to parse
+     *
+     * @return array The resulting typoscript array
+     */
+    protected function parseTS($string = '')
+    {
+        $parser = GeneralUtility::makeInstance(\TYPO3\CMS\Core\TypoScript\Parser\TypoScriptParser::class);
+        $parser->parse($string);
+        return $parser->setup;
+    }
+
+    /**
+     * Link string to the current page.
+     * @see \TYPO3\CMS\Frontend\Plugin\AbstractPlugin->pi_linkTP()
      *
      * @access public
      *
-     * @param string $content: HTML content to wrap in the div-tags with the class of the plugin
+     * @param string $str: The content string to wrap in <a> tags
+     * @param array $urlParameters: Array with URL parameters as key/value pairs
+     * @param bool $cache: Should the "no_cache" parameter be added?
+     * @param int $altPageId: Alternative page ID for the link.
      *
-     * @return string HTML content wrapped, ready to return to the parent object.
+     * @return string The input string wrapped in <a> tags
      */
-    public function pi_wrapInBaseClass($content)
+    public function pi_linkTP($str, $urlParameters = [], $cache = false, $altPageId = 0)
     {
-        if (!$GLOBALS['TSFE']->config['config']['disableWrapInBaseClass']) {
-            // Use class name instead of $this->prefixId for content wrapping because $this->prefixId is the same for all plugins.
-            $content = '<div class="tx-dlf-' . strtolower(Helper::getUnqualifiedClassName(get_class($this))) . '">' . $content . '</div>';
-            if (!$GLOBALS['TSFE']->config['config']['disablePrefixComment']) {
-                $content = "\n\n<!-- BEGIN: Content of extension '" . $this->extKey . "', plugin '" . Helper::getUnqualifiedClassName(get_class($this)) . "' -->\n\n" . $content . "\n\n<!-- END: Content of extension '" . $this->extKey . "', plugin '" . Helper::getUnqualifiedClassName(get_class($this)) . "' -->\n\n";
-            }
+        $conf = [];
+        if (!$cache) {
+            $conf['no_cache'] = true;
         }
-        return $content;
+        $conf['parameter'] = $altPageId ?: ($this->pi_tmpPageId ?: 'current');
+        $conf['additionalParams'] = $this->conf['parent.']['addParams'] . HttpUtility::buildQueryString($urlParameters, '&', true) . $this->pi_moreParams;
```

Comment:
buildQueryString() is unknown in TYPO3 8.7.

Reply:
Good catch! I'll fix it in a separate pull request.